### ISSUE TITLE
Piano roll grid enchancement

### ISSUE THREAD

**Issue by kinda-slow** [LMMS/lmms#1842]
It would be really great to have an option, which would allow you to have the grid "resolution" matched with the quantization value. What I mean, is that when you set quantization to 1/16 or 1/8, the notes you draw will be snapped to the vertical lines, that designate the boundaries between the notes of the corresponding length. So it will make much more sense just to hide other lines completely, since they are useless anyway and only clutter the view. Its even hard to tell where 2 and 4 are in the bar, because lines look almost the same. The other thing it would be great to have, is the ability to use 1/5, 1/7, 1/9 notes and their derivatives. Also it would be handy to have rows corresponding to black and white keys differ in color.

**Comment by softrabbit** [comment]
I tweaked something like this into the automation editor, but never got around to do the same to the piano roll. If someone feels the urge to do something about this issue, I suggest a look at https://github.com/LMMS/lmms/pull/426 for inspiration.

**Comment by Umcaruje** [comment]
I fixed most of this in #3062. The quintole and septole support is an issue with our quantisation and I feel like it could be a seperate issue.

**Comment by zonkmachine** [comment]
> The quintole and septole support is an issue with our quantisation and I feel like it could be a seperate issue.

:+1: to a separate 'tuplet' issue.

**Comment by z-up** [comment]
>The quintole and septole support is an issue with our quantisation and I feel like it could be a seperate issue.

So is there any chance that using quintoles and septoles will be possible in the future?

**Comment by zonkmachine** [comment]
> So is there any chance that using quintoles and septoles will be possible in the future?

Yes. Shouldn't be too complicated to implement but I haven't got the time to do it personally.

**Comment by RebeccaLaVie** [comment]
Seems like a separate issue should be created regarding the the quintole and septole support and this should be closed out or edited to only include the relevant information.

@z-up @Umcaruje

**Comment by musikBear** [comment]
@Umcaruje 
the layer-sequence in piano-roll looks flawed.
The bar-*shading* is placed *over* the notes.
Shading is lowest layer number. It ought to be notes.
If a note stretches over +1 bar the color of the note will be influenced
Note-Color over;
unshaded bar: #0C662E
shaded bar:   #0F6931
It results in difficulties determining the note-length, and is strangely tiresome for the eyes -it seams 'blurry'
![shadevunshade](https://user-images.githubusercontent.com/6662066/27514037-29e9d8a8-597e-11e7-8445-a28a3ff9f020.JPG)

**Comment by Umcaruje** [comment]
No, actually the bar shading is first drawn and the rest is drawn over it.
look at the bar grid color. what you're seeing is because the notes are
transparent and their background inevitably influences the final color
cause it gets mixed together. But no, shading is not drawn over the notes.

**Comment by musikBear** [comment]
ahhh yes they are transparent.. So that blur is a kind of 'moire' with two alpha-channels.. phff.. That would not be a simple fix
Tell me, do you also see that 'blurriness' in the shaded grid-part 'around' the note - to my lousy eyes its like there is a 'fatter' note inside the shading, like a 'bleed over' between the colors. If you (others) also see this, then the question could be, if the two colors (notes v. background) are too similar, and the notes maby should be complementary in respect to the background, eg swap green for *red*? -That would also need @RebeccaLaVie input, cause its her design

**Comment by RebeccaLaVie** [comment]
@musikBear I am open to tweaking the shades of colors and opacity in related to the grid if need be, but I have not heard of anyone else having difficulty with this. This might be one of those elements that you need to edit in the style.css individually.

**Comment by Spekular** [comment]
This is done in master, aside from the green vertical bars which are always displayed at 1/4th note positions.
